File: python/bronze/9506.py
while True:
    n = int(input())
    if n == -1:
        break
    divisor = [1]
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            divisor.append(i)
            if i != n // i:
                divisor.append(n // i)
    divisor.sort()
    if sum(divisor) == n:
        print(f"{n} = ", end="")
        for i in range(len(divisor)):
            print(divisor[i], end="")
            if i != len(divisor)-1:
                print(" + ", end="")
        print()
    else:
        print(f"{n} is NOT perfect.")

File: C++/bronze/9506.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
    int N, sum;
    vector<int> divisor;
    while(true){
        sum = 0;
        cin >> N;
        if(N == -1) break;
        divisor.clear(); // clear the vector
        for(int i = 1; i < N; i++){
            if(N % i == 0){
                divisor.push_back(i);
            }
        }
        for(unsigned int i = 0; i < divisor.size(); i++){
            sum += divisor[i];
        }
        if(sum == N){
            cout << N << " " << "=" << " ";
            for(unsigned int i = 0; i < divisor.size(); i++){
                cout << divisor[i];
                if(i != divisor.size() - 1) cout << " + "; // print + only between elements
            } 
        } else {
            cout << N << " is NOT perfect.";
        }
        cout << endl; // print newline after each iteration
    }
    return 0;
}
